src/helpers/renderHelpers.tsx

```diff
@@ -31,13 +32,15 @@ export function renderFirstIconOnly(children: React.ReactNode, shouldDisableColo
     // Check for an icon and if we find one make sure we only have 1 icon
     // total that we could render. We also need to display any set colors
     // so the component itself can style the color.
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/consistent-type-assertions
-    const childType = (child as any).type;
+    const childElement = child as React.ReactElement;
+    const childType = childElement.type as {displayName?: string};
```

Comment:
These casts are better than before. For the repeated ones (like `displayName`) can we add a type that we re-use?

I'm also not sure that all the casts to react elements are necessary in this file.